tests/BearSerializationTest.py

```diff
@@ -0,0 +1,34 @@
+import itertools
+import json
+import os
+import json
+import unittest
+
+from pyprint.NullPrinter import NullPrinter
+
+from coalib.bears.BEAR_KIND import BEAR_KIND
+from coalib.collecting.Collectors import collect_bears
+from coalib.output.JSONEncoder import create_json_encoder
+from coalib.output.printers.LogPrinter import LogPrinter
+
+
+class BearSerializationTest(unittest.TestCase):
+    """
+    Test to make sure all bears are serializable.
+    """
+
+    def setUp(self):
+        printer = LogPrinter(NullPrinter())
+        local_bears, global_bears = collect_bears(
+            [os.path.abspath("bears")],
+            ["**"],
+            [BEAR_KIND.LOCAL, BEAR_KIND.GLOBAL],
+            printer,
+            warn_if_unused_glob=False)
+        self.bears = list(itertools.chain(local_bears, global_bears))
+        self.JSONEncoder = create_json_encoder(use_relpath=False)
+
+    def test(self):
+        json.dumps(self.bears, cls=self.JSONEncoder)
+        # If no exception is thrown then the test passes
```

Comment:
Hmm alright. I think we go with latter one then. Catch the two exceptions.
Are there any more?
On 12 Jul 2016 6:27 pm, "Sheikh Araf" <notifications@github.com> wrote:

> In tests/BearSerializationTest.py
> <https://github.com/coala-analyzer/coala-bears/pull/574#discussion_r70431940>
> :
>
> > +    """
> > +
> > +    def setUp(self):
> > +        printer = LogPrinter(NullPrinter())
> > +        local_bears, global_bears = collect_bears(
> > +            [os.path.abspath("bears")],
> > +            ["**"],
> > +            [BEAR_KIND.LOCAL, BEAR_KIND.GLOBAL],
> > +            printer,
> > +            warn_if_unused_glob=False)
> > +        self.bears = list(itertools.chain(local_bears, global_bears))
> > +        self.JSONEncoder = create_json_encoder(use_relpath=False)
> > +
> > +    def test(self):
> > +        json.dumps(self.bears, cls=self.JSONEncoder)
> > +        # If no exception is thrown then the test passes
>
> The mechanism of serialization is written in coala. If we want to test
> unserializable bears here then we'll have to write mock bears. I wrote this
> test with the intention of not testing the mechanism, but to keep all bears
> in this repo serializable. What do you think?
>
> Secondly I guess we can catch the exceptions and neatly display the error,
> but then we'll need to do it exhaustively. e.g. TypeError is thrown when
> the object can't be serialized but there's also ValueError for circular
> reference.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/coala-analyzer/coala-bears/pull/574/files/5818022c8a7d2e6d652f0a63d36a6b343621728c#r70431940>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AHDgOYMmi6UVU1s3ak8gEgrvxtkby4K0ks5qU48rgaJpZM4JIza0>
> .
>